Jenkinsfile-test

```diff
@@ -0,0 +1,109 @@
+library("tdr-jenkinslib")
+
+def repo = "tdr-service-unavailable"
+def versionTag = "v${env.BUILD_NUMBER}"
+
+pipeline {
+  agent {
+    label "master"
+  }
+  stages {
+    stage("Run git secrets") {
+      steps {
+        script {
+          tdr.runGitSecrets(repo)
+        }
+      }
+    }
+    stage('Run tests') {
+      steps {
+        script {
+          sh "python3 -m venv venv"
+          sh "venv/bin/pip install -r requirements.txt"
+          sh "venv/bin/pip install -r requirements-test.txt"
+          sh "venv/bin/python -m pytest test"
+          sh "rm -r venv"
+        }
+      }
+    }
+
+    stage("Generate css") {
+      agent {
+        ecs {
+          inheritFrom "npm"
+        }
+      }
+      when {
+        expression { env.BRANCH_NAME == "main"}
+        beforeAgent true
+      }
+      steps {
+        script {
+          sh "npm ci"
+          sh "npm run build-css"
+          stash includes: "static/main.css", name: "main.css"
+        }
+      }
+    }
+
+    stage("Create deployment zip") {
+      when {
+        expression { env.BRANCH_NAME == "main"}
+        beforeAgent true
+      }
+      steps {
+        script {
+          unstash "main.css"
+          sh "./create_deployment_package.sh"
+          stash "deployment.zip"
+        }
+      }
+    }
+
+    stage("Upload deployment zip to s3") {
+      agent {
+        ecs {
+          inheritFrom "aws"
+          taskrole "arn:aws:iam::${env.MANAGEMENT_ACCOUNT}:role/TDRJenkinsNodeLambdaRoleIntg"
+        }
+      }
+      when {
+        expression { env.BRANCH_NAME == "main"}
+        beforeAgent true
+      }
+      steps {
+        script {
+          unstash "deployment.zip"
+          sh "aws s3 cp deployment.zip s3://tdr-backend-code-mgmt/${versionTag}/deployment.zip"
+
+          tdr.configureJenkinsGitUser()
+
+          sh "git tag ${versionTag}"
+          sshagent(['github-jenkins']) {
+            sh("git push origin ${versionTag}")
+          }
+        }
+        build(
+            job: "TDR Service Unavailable Deploy",
```

Comment:
Ooh, I wonder if we should rename this? It's just deploying the lambda, but when I just read it I assumed it actually switched to the service unavailable page!

What could we refer to the lambda (and therefore the deploy job) as? The service unavailable renderer? Not sure.

Reply:
I put it like this because all the other lambda deploy jobs are called `xxxx Deploy` and I called the job which switches the page over as `TDR Service Unavailable Run` as it matches what we've got for other lambdas like the database migrations. I see what you mean about it being ambiguous so I'm open to suggestions.

Reply:
Maybe it's enough to add a description to both Jenkins job making it clear that one deploys the Lambda code but doesn't affect the frontend load balancer, and the other switches between the frontend and service unavailable pages?

Reply:
I'll add the descriptions into the job. I don't think you can set it in the pipeline file, it's something that's done at the job level when you create the job, at least I can't find any of our other pipeline files that have a description.